From 124f151f28c0bfb94d0b91c67f3b9cd001c06cfc Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <chrisg@marvell.com>
Date: Thu, 5 Jan 2017 17:43:57 +0200
Subject: [PATCH 0770/1345] dts: a70x0: add device tree configuration for
 board setup B

commit  eca1be6de9f02d0f40fdb4ae0f0c27766311b74c from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

A7040 DB board is a modular board,
which supports multiple possible configurations.
This patch introduce predefined board topology for setup B.

- Add device tree for B setup
	- AP_UART, AP_SD
	- CP0_RGMII0, RGMII1, CP0_SDIO, CP0_I2C,
	  CP0_SATA1, CP0_SPI

Change-Id: I1900dfa58d4f02a68bc5f72487bd31c69d72cf15
Signed-off-by: Christine Gharzuzi <chrisg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35363
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../bindings/arm/marvell/armada-7k-8k.txt          |    3 +
 .../mvebu/a7k-a8k/armada7040-db-setup.txt          |  114 +++++++------
 arch/arm64/boot/dts/marvell/Makefile               |    1 +
 arch/arm64/boot/dts/marvell/armada-7040-db-B.dts   |  176 ++++++++++++++++++++
 4 files changed, 246 insertions(+), 48 deletions(-)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-7040-db-B.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index 3819941..a35c6e1 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -23,6 +23,9 @@ Boards:
    - "marvell,armada7040-db-A", "marvell,armada7040-db"
       when the board being used is the Armada 7040 Development board
 
+   - "marvell,armada7040-db-B", "marvell,armada7040-db"
+      when the board being used is the Armada 7040 Development board
+
    - "marvell,armada7040-db-nand", "marvell,armada7040-db"
       when the board being used is the Armada 7040 Development board
 
diff --git a/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt b/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
index 210489c..8e70fc7 100644
--- a/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
+++ b/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
@@ -21,19 +21,6 @@ Board Setup
 
 For A7040 A revision SoC - Please use armada-7040-db-A configuration,
 
-|		| 		1: A	|
-|---------------|-----------------------|
-|Device tree	|armada-7040-db-A.dtb	|
-|CP0-SW1:1	| ON			|
-|CP0-SW1:2	| OFF			|
-|CP0-SW1:3	| ON			|
-|CP0-SW1:4	| OFF			|
-|CP0-SW1:5	| OFF			|
-|CP0-SW1:6	| ON			|
-|CP0-SW1:7	| ON			|
-|CP0-SW1:8	| OFF			|
------------------------------------------
-
 DPR Setting
 ------------
 |					| 	1: A				 |
@@ -45,20 +32,44 @@ DPR Setting
 | 					| DPR2,3,4,5 - 1-2 position		 |
 ----------------------------------------|----------------------------------------|
 
+|		| 	1: A		|	2:B		|
+|---------------|-----------------------|-----------------------|
+|Device tree	|armada-7040-db-A.dtb	|armada-7040-db-B.dtb	|
+|CP0-SW1:1	| ON			| OFF			|
+|CP0-SW1:2	| OFF			| ON			|
+|CP0-SW1:3	| ON			| ON			|
+|CP0-SW1:4	| OFF			| ON			|
+|CP0-SW1:5	| OFF			| ON			|
+|CP0-SW1:6	| ON			| ON			|
+|CP0-SW1:7	| ON			| ON			|
+|CP0-SW1:8	| OFF			| OFF			|
+-----------------------------------------------------------------
+
+DPR Setting
+------------
+|					| 	1: A				 |	2: B				  |
+|---------------------------------------|----------------------------------------|----------------------------------------|
+| Mpp Jumper Module (SLM1548)		| Default release (ALL DPRs 1-2 position)| Default release (ALL DPRs 1-2 position)|
+| Serdess Jumper Module (SLM1547)	| DPR152,153,154,155 - 1-2 position	 | DPR152,153,154,155 - 1-2 position      |
+|					| DPR148,149,150,151 - 2-3 position	 | DPR148,149,150,151 - 1-2 position      |
+| CP0 Serdess module (SLM1521)		| SW1 - (1,3,6,7)= ON,  (2,4,5,8)=OFF	 | SW1 - (2,3,4,5,6,7)= ON,  (1,8)=OFF    |
+| 					| DPR2,3,4,5 - 1-2 position		 | DPR2,3,4,5 - 2-3 position              |
+----------------------------------------|----------------------------------------|----------------------------------------|
+
 The tables below summarizes the interface configuration of each setup
 
 SerDes Configuration
 ------------------------
 
-| CP0 Lane 	| 	1: A		|
-|---------------|-----------------------|
-| 0		| SGMII2		|
-| 1		| USB3_HOST0		|
-| 2		| KR (10G)		|
-| 3		| SATA1			|
-| 4		| USB3_HOST1		|
-| 5		| PCIE2 (x1)		|
------------------------------------------
+| CP0 Lane 	| 	1: A		|	2: B		|
+|---------------|-----------------------|-----------------------|
+| 0		| SGMII2		|	PICE0 (x4)	|
+| 1		| USB3_HOST0		|	PCIE0 (x4)	|
+| 2		| KR (10G)		|	PCIE0 (x4)	|
+| 3		| SATA1			|	PCIE0 (x4)	|
+| 4		| USB3_HOST1		|	USB3_HOST1	|
+| 5		| PCIE2 (x1)		|	PCIE2 (x1)	|
+-----------------------------------------------------------------
 
 - USB2_H0 (UTMI only)
 - USB2_H1 (UTMI & USB3 SerDes)
@@ -66,30 +77,30 @@ SerDes Configuration
 
 Multi-purpose pin configurations
 --------------------------------
-Setup  #	|	1	|
-Name   #	|  	A 	|
----------------------------------
-AP806:				|
----------------------------------
-	AP-SDIO | [0-5]		|
-	AP-UART0| [11,19]	|
----------------------------------
-CP:				|
----------------------------------
-	CP-SPI0	| N/C		|
-	CP-SPI1	| [13-16]	|
-	NAND	| N/C		|
-	RGMII0	| N/C		|
-	RGMII1	| [0-11]	|
-	CP-UART0| [29-30]	|
-	CP-UART1| N/C		|
-	SD	| [56-61]	|
-	GPIO	| [62]		|
-	TDM	| N/C		|
-	TWSI	| [37-38]	|
-	SATA0	| N/C		|
-	SATA1	| [28]		|
----------------------------------
+Setup  #	|	1	|	2	|
+Name   #	|  	A 	|	B	|
+-------------------------------------------------
+AP806:				|		|
+-------------------------------------------------
+	AP-SDIO | [0-5]		| [0-5]		|
+	AP-UART0| [11,19]	| [11,19]	|
+-------------------------------------------------
+CP:				|		|
+-------------------------------------------------
+	CP-SPI0	| N/C		| N/C		|
+	CP-SPI1	| [13-16]	| [13-16]	|
+	NAND	| N/C		| N/C		|
+	RGMII0	| N/C		| [44-55]	|
+	RGMII1	| [0-11]	| [0-11]	|
+	CP-UART0| [29-30]	| [29-30]	|
+	CP-UART1| N/C		| N/C		|
+	SD	| [56-61]	| [56-61]	|
+	GPIO	| [62]		| [62]		|
+	TDM	| N/C		| N/C		|
+	TWSI	| [37-38]	| [37-38]	|
+	SATA0	| N/C		| N/C		|
+	SATA1	| [28]		| [28]		|
+-------------------------------------------------
 
 Network configuration
 ---------------------
@@ -97,6 +108,13 @@ Network configuration
  Interface	| CP#	| PPv2 Port	| GMAC Port	| Board Interface		|
  ---------------|-------|---------------|---------------|-------------------------------|
  eth0		| CP0	| eth0		| 0		| SFI (10G)			|
- eth1		| CP0	| eth1		| 2		| RGMII1			|
- eth2		| CP0	| eth2		| 3		| RGMII2			|
+ eth1		| CP0	| eth1		| 2		| RGMII0			|
+ eth2		| CP0	| eth2		| 3		| RGMII1			|
+ ----------------------------------------------------------------------------------------
+
+2: B:
+ Interface      | CP#   | PPv2 Port     | GMAC Port     | Board Interface               |
+ ---------------|-------|---------------|---------------|-------------------------------|
+ eth0           | CP0   | eth1          | 2             | RGMII0                        |
+ eth1           | CP0   | eth2          | 3             | RGMII1                        |
  ----------------------------------------------------------------------------------------
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 304cf58..f9bd457 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -11,6 +11,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-37x0-customer.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-espressobin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-A.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-nand.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
new file mode 100644
index 0000000..c027b43
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
@@ -0,0 +1,176 @@
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 7040 development board
+ * This board file supports the B configuration of the board
+ */
+
+#include "armada-7040-db.dtsi"
+
+/ {
+	model = "Marvell Armada-7040 development board B setup";
+	compatible = "marvell,armada7040-db-B", "marvell,armada7040-db", "marvell,armada7040",
+			"marvell,armada-ap806-quad", "marvell,armada-ap806";
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	chosen { };
+
+	ap806 {
+		config-space {
+			sdhci@6e0000 {
+				status = "okay";
+			};
+			serial@512000 {
+				status = "okay";
+			};
+			i2c@511000 {
+				status = "disabled";
+				clock-frequency = <100000>;
+			};
+		};
+	};
+
+	cpn-110-master {
+		config-space {
+			serial@702000 {
+				status = "disabled";
+			};
+			serial@702100 {
+				status = "disabled";
+			};
+			serial@702200 {
+				status = "disabled";
+			};
+			serial@702300 {
+				status = "disabled";
+			};
+			sata: sata@540000 {
+				status = "disabled";
+
+				sata-port@0 {
+					status = "disabled";
+				};
+				sata-port@1 {
+					status = "disabled";
+				};
+			};
+			usb3h0: usb3@500000 {
+				status = "disabled";
+			};
+			usb3h1: usb3@510000 {
+				status = "okay";
+			};
+			u3d@520000 {
+				status = "disabled";
+			};
+			udc@524100 {
+				status = "disabled";
+			};
+			sdhci@780000 {
+				status = "okay";
+			};
+			spi@700680 {
+				status = "okay";
+			};
+			i2c1: i2c@701000 {
+				status = "okay";
+				clock-frequency = <100000>;
+			};
+			i2c2: i2c@701100 {
+				status = "disabled";
+			};
+
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy2: ethernet-phy@2 {
+					reg = <0>;
+				};
+				phy3: ethernet-phy@3 {
+					reg = <1>;
+				};
+			};
+			gop {
+				emac0: mac0 {
+					/*disabled*/
+				};
+				emac2: mac2 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&phy2>;
+				};
+				emac3: mac3 {
+					phy-mode = "rgmii"; /* rgmii-2 */
+					phy = <&phy3>;
+				};
+			};
+			ppv22@000000 {
+				eth0: eth0@010000 {
+					status = "disabled";
+				};
+				eth1: eth1@020000 {
+					status = "okay";
+				};
+				eth2: eth2@030000 {
+					status = "okay";
+				};
+			};
+			eip197: eip197@800000 {
+				status = "okay";
+			};
+			comphy {
+				status = "okay";
+			};
+		};
+		pcie@0x600000 {
+			status = "okay";
+			num-lanes = <4>;
+		};
+		pcie@0x620000 {
+			status = "disabled";
+		};
+		pcie@0x640000 {
+			status = "okay";
+		};
+	};
+};
-- 
1.7.9.5

